### PR TITLE
Progress bar changes color on completion and failure

### PR DESCRIPTION
**Pull request by ketan96-m**
### Issue
https://github.com/ssl-hep/ServiceX_frontend/issues/382

### Description
The current progress bar is green through out the process.
Introduced a muted green color for normal process and added a brighter green color when transform bar and download bar finishes all files.
If the transformation fails prematurely, then the bar changes the color to red.

Normal progress:
image

Completed progress:
image

Failed Progress:
image

### Solution:
Created a custom `TransformStatusProgress` class which inherits the `Progress` from `rich.progress`.
`get_renderables()` function is overloaded to update the `BarColumn()` with the `red` color in case of failed files.

Added an if condition inside the `query.py`'s `transform_status_listener` that checks that the transform is complete and if there are any failed files and then call the update method for the progress bar.

The update method takes a new keyword argument to indicate if transformation is complete or not.

Added subsequent test cases to test this new `get_renderables()` function and also edited other test cases to with the new `TransformStatusProgress`.

### CODE REVIEW

File: servicex/expandable_progress.py [hunk 1]
```diff
@@ -183,3 +184,16 @@ def advance(self, task_id, task_type):
...
             self.progress.advance(task_id=task_id)
+
+
+class TranformStatusProgress(Progress):
+    def get_renderables(self):
+        for task in self.tasks:
+            if task.fields.get("bar") == "failure":
+                self.columns = [
+                    TextColumn("[progress.description]{task.description}"),
+                    BarColumn(complete_style="rgb(255,0,0)"),
```
Comment (BenGalewsky) [inline servicex/expandable_progress.py:195]: Why is this here and different from `complete_style="rgb(114,156,31)` above?

File: servicex/expandable_progress.py [hunk 2]
```diff
@@ -74,9 +74,10 @@ def __init__(self,
         self.progress_counts = {}
         if display_progress:
             if self.overall_progress or not provided_progress:
-                self.progress = Progress(
+                self.progress = TranformStatusProgress(
                     TextColumn("[progress.description]{task.description}"),
-                    BarColumn(),
+                    BarColumn(complete_style="rgb(114,156,31)",
```
Comment (ketan96-m) [inline servicex/expandable_progress.py:79]: This is the muted green color when the bar is progressing.
image

File: servicex/expandable_progress.py (hunk 1 above)
Comment (ketan96-m) [reply servicex/expandable_progress.py:195]: And this red color is for when the bar is progressing but some files have failed.

**Comment by ponyisi** [comment]
Superseded by #408
